process/block/preprocess/intermediateResults.go

```diff
@@ -93,7 +94,13 @@ func (irp *intermediateResultsProcessor) CreateAllInterMiniBlocks() map[uint32]*
 				return bytes.Compare(miniBlocks[i].TxHashes[a], miniBlocks[i].TxHashes[b]) < 0
 			})
 
+			mbSpaceRemained := int(maxMbSpaceRemained) - mbAdded
```

Comment:
in case of intermediate processor you should add all the created miniblocks. otherwise it will break the whole logic of smart contract results and rewardProcessor. Everything which was created during this round, has to be added here, no exception. The next round you do not know about this data.

Reply:
Fixed